packages/labs/task/src/task.ts

```diff
@@ -233,7 +234,9 @@ export class Task<T extends [...unknown[]] = any, R = any> {
     return this._error;
   }
 
-  render(renderer: StatusRenderer<R>) {
+  render<SR extends StatusRenderer<R, E>>(
+    renderer: Partial<SR>
+  ): ReturnType<SR[this['status']]> | undefined {
```

Comment:
I think this is suboptimal but I think I didn't manage better with type assertions. If I understand correctly this means that `undefined` is always a possible return value even when all 4 functions are defined and don't return `undefined`

Reply:
Don't understand though why in your screenshot above this is not the case.

Reply:
Why do we need the `| undefined`?

Reply:
A partial record of callbacks will always return `| undefined` as a possible type. If it were a partial record of numbers it'd be the same: `number | undefined`

Reply:
And there we go, we don't need it. Updated.